### ISSUE TITLE
Rustup 1.2.0 fails on FreeBSD 11.0-RELEASE

### ISSUE THREAD

**Issue by pwrdwnsys** [rust-lang/rustup#1044]
Rustup is no longer working since the self-update.

```
$ rustup update
info: syncing channel updates for 'stable-x86_64-unknown-freebsd'
error: could not download file from 'https://static.rust-lang.org/dist/channel-rust-stable.toml.sha256' to '/home/pwrdwnsys/.rustup/tmp/qsdrj7fv1vgemm2x_file'
info: checking for self-updates
error: could not download file from 'https://static.rust-lang.org/rustup/release-stable.toml' to '/tmp/rustup-update.3geQN1MvXagB/release-stable.toml'
info: caused by: error during download
info: caused by: [1] Unsupported protocol (Protocol "https" not supported or disabled in libcurl)

$ rustup --version
rustup 1.2.0 (70faf07 2017-04-08)

$ rustup show
Default host: x86_64-unknown-freebsd

stable-x86_64-unknown-freebsd (default)
rustc 1.16.0 (30cf806ef 2017-03-10)

$ uname -a
FreeBSD git 11.0-RELEASE-p8 FreeBSD 11.0-RELEASE-p8 #0: Wed Feb 22 06:12:04 UTC 2017     root@amd64-builder.daemonology.net:/usr/obj/usr/src/sys/GENERIC  amd64
```

I also did a clean install of FreeBSD 11-RELEASE, and rustup would not perform a new install either.

```
$ curl https://sh.rustup.rs -sSf | sh
info: downloading installer

Welcome to Rust!

This will download and install the official compiler for the Rust programming 
language, and its package manager, Cargo.

It will add the cargo, rustc, rustup and other commands to Cargo's bin 
directory, located at:

  /home/pwrdwnsys/.cargo/bin

This path will then be added to your PATH environment variable by modifying the
profile file located at:

  /home/pwrdwnsys/.profile

You can uninstall at any time with rustup self uninstall and these changes will
be reverted.

Current installation options:

   default host triple: x86_64-unknown-freebsd
     default toolchain: stable
  modify PATH variable: yes

1) Proceed with installation (default)
2) Customize installation
3) Cancel installation


info: syncing channel updates for 'stable-x86_64-unknown-freebsd'
error: could not download file from 'https://static.rust-lang.org/dist/channel-rust-stable.toml.sha256' to '/home/pwrdwnsys/.rustup/tmp/elle0cwj6z9ug13s_file'
info: caused by: error during download
info: caused by: [1] Unsupported protocol (Protocol "https" not supported or disabled in libcurl)

$ rustup --version
rustup 1.2.0 (70faf07 2017-04-08)

$ pkg info curl
curl-7.53.1_1
Name           : curl
Version        : 7.53.1_1
Installed on   : Sun Apr  9 00:30:48 2017 UTC
Origin         : ftp/curl
Architecture   : FreeBSD:11:amd64
Prefix         : /usr/local
Categories     : www ftp net ipv6
Licenses       : MIT
Maintainer     : sunpoet@FreeBSD.org
WWW            : http://curl.haxx.se/
Comment        : Non-interactive tool to get files from FTP, GOPHER, HTTP(S) servers
Options        :
	CARES          : off
	CA_BUNDLE      : on
	COOKIES        : on
	CURL_DEBUG     : off
	DEBUG          : off
	DOCS           : on
	EXAMPLES       : on
	GNUTLS         : off
	GSSAPI_BASE    : on
	GSSAPI_HEIMDAL : off
	GSSAPI_MIT     : off
	GSSAPI_NONE    : off
	HTTP2          : on
	IDN            : off
	IPV6           : on
	LDAP           : off
	LDAPS          : off
	LIBSSH2        : off
	METALINK       : off
	NSS            : off
	OPENSSL        : on
	POLARSSL       : off
	PROXY          : on
	PSL            : off
	RTMP           : off
	SMB            : off
	THREADED_RESOLVER: on
	TLS_SRP        : on
	WOLFSSL        : off
Shared Libs required:
	libnghttp2.so.14
Shared Libs provided:
	libcurl.so.4
Annotations    :
	cpe            : cpe:2.3:a:haxx:curl:7.53.1:::::freebsd11:x64:1
	repo_type      : binary
	repository     : FreeBSD
Flat size      : 3.04MiB
Description    :
curl is a client to get documents/files from servers, using any of the
supported protocols. The command is designed to work without user
interaction or any kind of interactivity.

curl offers a busload of useful tricks like proxy support, user
authentication, ftp upload, HTTP post, SSL (https:) connections, file
transfer resume and more.

WWW: http://curl.haxx.se/

$ openssl version
OpenSSL 1.0.2k-freebsd  26 Jan 2017

$ uname -a
FreeBSD fbsdrust 11.0-RELEASE-p8 FreeBSD 11.0-RELEASE-p8 #0: Wed Feb 22 06:12:04 UTC 2017     root@amd64-builder.daemonology.net:/usr/obj/usr/src/sys/GENERIC  amd64
```

**Comment by reezer** [comment]
Having the same issue.

HTTPS is supported. My guess is that the binary is compiled with an old version of curl, which brings incompatibility.

```
curl 7.53.1 (amd64-portbld-freebsd11.0) libcurl/7.53.1 LibreSSL/2.5.2 zlib/1.2.11 nghttp2/1.21.0
Protocols: dict file ftp ftps gopher http https imap imaps pop3 pop3s rtsp smtp smtps telnet tftp 
Features: AsynchDNS IPv6 Largefile NTLM NTLM_WB SSL libz HTTP2 UnixSockets HTTPS-proxy
```

**Comment by tatsuya6502** [comment]
Just ran into the same issue here. Is there any way to install an old version of rustup?

**Comment by tatsuya6502** [comment]
This looks like a duplicate of #1051.

**Comment by pwrdwnsys** [comment]
@tatsuya6502 I have not found a way to use an old version of rustup. The version of nightly Rust in FreeBSD Ports is 1.17.0 and last updated in early March - I don't think this port is getting much update work since rustup had esentially taken over.

**Comment by tatsuya6502** [comment]
@pwrdwnsys Thanks for the info. OK. I will wait for the fix of #1051.

**Comment by tatsuya6502** [comment]
I realized I can build `rustup-init` by myself. I got a working `rustup` now. https://github.com/rust-lang-nursery/rustup.rs/issues/1051#issuecomment-294081620

**Comment by pwrdwnsys** [comment]
Hopefully this will be fixed by #1065

**Comment by Diggsey** [comment]
Closing in favour of #1051 since it has more discussion.
